src/dbusextendedabstractinterface.cpp

```diff
@@ -0,0 +1,440 @@
+// -*- c++ -*-
+
+/*!
+ *
+ * Copyright (C) 2015 Jolla Ltd.
+ *
+ * Contact: Valerio Valerio <valerio.valerio@jolla.com>
+ * Author: Andres Gomez <andres.gomez@jolla.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+
+#include "dbusextendedpendingcallwatcher_p.h"
+
+#include <DBusExtendedAbstractInterface>
+
+#include <QtDBus/QDBusMetaType>
+#include <QtDBus/QDBusMessage>
+#include <QtDBus/QDBusPendingCall>
+#include <QtDBus/QDBusPendingCallWatcher>
+#include <QtDBus/QDBusPendingReply>
+
+#include <QtCore/QDebug>
+#include <QtCore/QMetaProperty>
+
+
+Q_GLOBAL_STATIC_WITH_ARGS(QByteArray, dBusPropertiesInterface, ("org.freedesktop.DBus.Properties"))
+Q_GLOBAL_STATIC_WITH_ARGS(QByteArray, dBusPropertiesChangedSignal, ("PropertiesChanged"))
+Q_GLOBAL_STATIC_WITH_ARGS(QByteArray, propertyChangedSignature, ("propertyChanged(QString,QVariant)"))
+Q_GLOBAL_STATIC_WITH_ARGS(QByteArray, propertyInvalidatedSignature, ("propertyInvalidated(QString)"))
+
+
+DBusExtendedAbstractInterface::DBusExtendedAbstractInterface(const QString &service, const QString &path, const char *interface, const QDBusConnection &connection, QObject *parent)
+    : QDBusAbstractInterface(service, path, interface, connection, parent)
+    , m_sync(false)
+    , m_useCache(false)
+    , m_getAllPendingCallWatcher(0)
+    , m_lastExtendedError()
```

Comment:
Mmm ... the default constructor is already OK. It will construct an error with the default "None" value.

Do you see any improvement by passing a 0?